packages/flutter/lib/src/widgets/text.dart

```diff
@@ -769,50 +761,20 @@ class _RichText extends StatelessWidget {
   final StrutStyle? strutStyle;
   final TextWidthBasis textWidthBasis;
   final ui.TextHeightBehavior? textHeightBehavior;
-  final Color? selectionColor;
-
-  @override
-  Widget build(BuildContext context) {
-    final SelectionRegistrar? registrar = SelectionContainer.maybeOf(context);
-    return RichText(
-      key: textKey,
-      textAlign: textAlign,
-      textDirection: textDirection,
-      locale: locale,
-      softWrap: softWrap,
-      overflow: overflow,
-      textScaler: textScaler,
-      maxLines: maxLines,
-      strutStyle: strutStyle,
-      textWidthBasis: textWidthBasis,
-      textHeightBehavior: textHeightBehavior,
-      selectionRegistrar: registrar,
-      selectionColor: selectionColor,
-      text: text,
-    );
-  }
-}
-
-class _SelectableTextContainer extends StatefulWidget {
-  const _SelectableTextContainer({
-    required this.textKey,
-    required this.child,
-  });
-
-  final GlobalKey textKey;
-  final Widget child;
+  final Color selectionColor;
 
   @override
   State<_SelectableTextContainer> createState() => _SelectableTextContainerState();
 }
 
 class _SelectableTextContainerState extends State<_SelectableTextContainer> {
   late final _SelectableTextContainerDelegate _selectionDelegate;
+  final GlobalKey _textKey = GlobalKey();
```

Comment:
Ok so this is now created here on State instead of in the ` build` method like before. Looks good.